### PR TITLE
Add support for ufw

### PR DESCRIPTION
**Pull request by EricLin0509**

### CODE REVIEW

File: apparmor.d/profiles-s-z/ufw [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# Copyright (C) 2024 EricLin <ericlin050914@gmail.com>
+# SPDX-License-Identifier: GPL-2.0-only
+
+abi <abi/3.0>,
+
+include <tunables/global>
+
+@{exec_path} = @{bin}/ufw
+profile ufw @{exec_path} {
+
```
Comment (roddhjav) [inline apparmor.d/profiles-s-z/ufw:11]: Remove this empty line

File: apparmor.d/profiles-s-z/ufw [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    capability net_admin,
+
+    network netlink raw,
+    network inet dgram,
+    network inet6 dgram,
+
+    @{exec_path} mr,
+
+    @{bin}/ r,
+    @{bin}/python3* ix,
```
Comment (roddhjav) [inline apparmor.d/profiles-s-z/ufw:24]: If you need this, it probably means you also need the python abstraction.

File: apparmor.d/profiles-s-z/ufw [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    @{bin}/cat ix,
+    @{bin}/xtables-nft-multi ix,
+
+    @{lib}/ufw/ufw-init ix,
+
+    @{PROC}/@{pid}/stat r,
+    @{PROC}/@{pid}/fd/ r,
+    @{PROC}/@{pid}/net/ip_tables_names r,
```
Comment (roddhjav) [inline apparmor.d/profiles-s-z/ufw:32]: Move these rules before the `/dev/` rules.

File: apparmor.d/profiles-s-z/ufw [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    owner @{bin}/env r,
```
Comment (roddhjav) [inline apparmor.d/profiles-s-z/ufw:34]: That should be: `@{bin}/env r,` and it should be moved with the other `@{bin}` rules.

File: apparmor.d/profiles-s-z/ufw [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+    /etc/ufw/{,**} rwk,
+
+    /etc/default/ufw r,
+
+    /run/ufw.lock wk,
```
Comment (roddhjav) [inline apparmor.d/profiles-s-z/ufw:40]:
```suggestion
    @{run}/ufw.lock rwk,
```

File: apparmor.d/profiles-s-z/ufw [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+    /etc/gai.conf r,
+    /etc/nsswitch.conf r,
+    /etc/passwd r,
+    /etc/services r,
```
Comment (roddhjav) [inline apparmor.d/profiles-s-z/ufw:45]: Remove these rules and add the `nameservice-strict` abstraction.

**Comment by roddhjav** [comment]
Thanks a lot for this PR. I made a few minor comments, once they are integrated, the PR will be ready to be merged.

**Comment by EricLin0509** [comment]
Sorry for the late reply, it should be fixed now.

**Comment by roddhjav** [comment]
Thanks, merged.

**Pull request merged**